InfiniteLoopBookProject/src/com/st/lms/menu/Menu.java

```diff
@@ -0,0 +1,420 @@
+								System.out.println("Please enter the publisher's address.");
+								String publisherAddr = scan.nextLine();
+								System.out.println("Please enter the publisher's phone number.");
+								long phone = Long.parseLong(getValidPhone());
+								Service.addPublisher(publisherID, publisherName, publisherAddr, phone);
+							}
+							break;
+						default:
+							break;
+					}
+				}
+				break;
+			case EXIT:
+				System.out.println("Returning to main menu...");
+				input = EXIT; //failsafe in case input was modified
+				break;
+			default:
+				break;
+			}
+		}while(input != EXIT);
+		run();
+	}
+	
+	private void AuthorMenu() {
+		List<String> options = new ArrayList<String>();
+		int authorID, input;
+		String authorName;
+		options.add("1) Display current list of authors");
+		options.add("2) Add a new author");
+		options.add("3) Remove a author");
+		options.add("4) Search for an author");
+		options.add("5) Update an author");
+		options.add("" + EXIT + ") Return to main menu");
+		
+		do {
+			System.out.println();
+			System.out.println();
+			System.out.println("Author Menu. Please select one of the options below.");
+			for(String option : options) {
+				System.out.println(option);
+			}
+			input = getValidOption(options.size());
+			
+			switch(input) {
+			case 1:
+				Service.printAuthors();
+				break;
+			case 2:
+				System.out.println("Please enter the author's id number.");
+				authorID = getValidID();
+				System.out.println("Please enter the author's name.");
+				authorName = scan.nextLine();
+				Service.addAuthor(authorID, authorName);
+				System.out.println("The database was successfully updated!");
+				break;
+			case 3:
+				System.out.println("Please enter the id of the book you want to remove!");
+				authorID = getValidID();
+				Service.removeAuthor(authorID);
+				System.out.println("The book was removed from the database!");
+				break;
+			case 4:
+				System.out.println("Please enter the id number.");
+				authorID = getValidID();
+				Author author = Service.getAuthor(authorID);
+				if(author == null)
+					System.out.println("The book's author does not exist in the database.");
+				else {
+					System.out.println("Author ID: " + author.getID()
+							+ "\nAuthor's Name: " + author.getName());
+				}
+				break;
+			case 5:
+				System.out.println("Please enter the id number of the author you would like to update.");
+				authorID = getValidID();
+				author = Service.getAuthor(authorID);
+				if(author==null)
+					System.out.println("That author does not exist in our records!");
+				else {
+						System.out.println("Please enter the author's new name.");
+						authorName = scan.nextLine();
+						author = new Author(authorName, authorID);
+						Service.updateAuthor(authorID, author);
+					}
+			case EXIT:
+				System.out.println("Returning to main menu...");
+				break;
+			default:
+				break;
+			}
+		}while(input != EXIT);
+		run();

```

Comment:
As I said in my review of the earlier PR, having the main menu call a sub-menu which calls the main menu which calls ... (and so on) is a sloppy, at best, architectural design. A perhaps clearer explanation of my concern is this: Each method, from `main()` on down, is a series of statements to do *something*. It can accomplish that purpose by calling other methods, or even (as in the case of something like the factorial function) calling itself, but it should carry out its purpose and then return. Calling the method that called this one because you want its logic to start over again is a dangerous (at least if someone were to either run this code in an embedded context with a severely limited stack or feed it input from an automated script) hack for the problem that loops were designed to solve.